### PR TITLE
[FX-933] Add support for single ForageError

### PR DESCRIPTION
**Pull request by djoksimo**
## What

ℹ️ The new `ForageError` format also orders the parameters in order of usage popularity:

1. code
2. httpStatusCode
3. message
4. details

- [x] Add support for `ForageError`
- [x] Mark `forageError.errors` as deprecated
- [x] Update tests to cover both paths
- [x] Add more `ForageError` coverage in the sample app and log `ForageError` 
- [x] Updated README to document Forage error handling

## Why

https://linear.app/joinforage/issue/FX-933/ios-support-single-error-for-client-facing-errors

## Test Plan

- ✅ Updated unit tests
- ✅ Updated sample app
- ✅ Did some manual verification across edge cases like forage errors with `details` (ebt_error_51)

## How

Have to bump `MINOR` version level of the associated release.

### CODE REVIEW

File: Sources/ForageSDK/Foundation/Network/Provider.swift [hunk 1]
```diff
@@ -146,17 +146,30 @@ class Provider {
 
     private func processData<T: Decodable>(model: T.Type, data: Data?, response: HTTPURLResponse?, error: Error?, completion: @escaping (Result<T, Error>) -> Void) {
         guard let data = data else {
-            return completion(.failure(ForageError(errors: [ForageErrorObj(httpStatusCode: response?.statusCode ?? 500, code: "invalid_data", message: "Invalid Data")])))
+            return completion(.failure(ForageError.create(
+                code: "invalid_input_data",
+                httpStatusCode: response?.statusCode ?? 500,
+                message: "Double check the reference documentation to validate the request body, and scan your implementation for any other errors."
+            )))
```
Comment (djoksimo) [inline Sources/ForageSDK/Foundation/Network/Provider.swift:153]: Updated to [more accurate/documented error code](https://docs.joinforage.app/reference/errors#invalid_input_data)

**Review by dleis612 — CHANGES_REQUESTED** [review]

File: Sources/ForageSDK/Foundation/Network/Model/ForageErrorModel.swift [hunk 2]
```diff
@@ -26,23 +26,43 @@ struct ForageErrorSource: Codable {
...
 /// Represents an error that occurs when a request to submit a `ForageElement` to the Forage API fails.
 public struct ForageError: Error, Codable {
+    /// A short string explaining why the request failed. The error code string corresponds to the HTTP status code.
+    /// [Learn more about SDK error codes](https://docs.joinforage.app/reference/errors#code-and-message-pairs-1)
+    public let code: String
+
+    /// The HTTP status that the Forage API returns in response to the request.
+    public let httpStatusCode: Int
+
+    /// A developer-facing message about the error, not to be displayed to customers.
```
[the remark this answers is not in the document]
Comment (djoksimo) [reply Sources/ForageSDK/Foundation/Network/Model/ForageErrorModel.swift]: Good catch! ccc8dda2c8e85f91d334627277dd1c8110706120

File: Sources/ForageSDK/Foundation/Network/Model/ForageErrorModel.swift [hunk 3]
```diff
@@ -26,23 +26,43 @@ struct ForageErrorSource: Codable {
 
 /// Represents an error that occurs when a request to submit a `ForageElement` to the Forage API fails.
 public struct ForageError: Error, Codable {
+    /// A short string explaining why the request failed. The error code string corresponds to the HTTP status code.
```
[the remark this answers is not in the document]
Comment (djoksimo) [reply Sources/ForageSDK/Foundation/Network/Model/ForageErrorModel.swift]: ccc8dda2c8e85f91d334627277dd1c8110706120

**Review by dleis612 — APPROVED** [review]
